app/controllers/responses_controller.rb

```
@@ -0,0 +1,196 @@
+      render json: { error: @response.errors.full_messages.to_sentence }, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /responses/:id/unsubmit
+  # Instructor/Admin allows a submitted response to be reopened for editing.
+  def unsubmit
+    return render json: { error: "#{response_label} not found" }, status: :not_found unless @response
+
+    if @response.is_submitted?
+      @response.update(is_submitted: false)
+      render json: { message: "#{response_label} reopened for editing. The reviewer can now make changes.", response: @response }, status: :ok
```

RuboCop (1.86.1)

[convention] Line is too long. [142/120]

(Layout/LineLength)